server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -287,10 +291,10 @@ private void allocateUnassigned() {
                                 case THROTTLE -> {
                                     nodeIdsIterator.wasThrottled = true;
                                     unallocatedStatus = AllocationStatus.DECIDERS_THROTTLED;
-                                    logger.trace("Couldn't assign shard [{}] to [{}]: {}", shard.shardId(), nodeId, decision);
+                                    logger.debug("Couldn't assign shard [{}] to [{}]: {}", shard.shardId(), nodeId, decision);
                                 }
                                 case NO -> {
-                                    logger.trace("Couldn't assign shard [{}] to [{}]: {}", shard.shardId(), nodeId, decision);
+                                    logger.debug("Couldn't assign shard [{}] to [{}]: {}", shard.shardId(), nodeId, decision);
```

Comment:
This may log meaningful information when constant decision with a message is returned.
Additionally we might do `allocation.setDebugMode(EXCLUDE_YES_DECISIONS)` if `logger.isTraceEnabled()`

Reply:
Additionally (not sure if it was considered before) but we may order allocation deciders so that computationally cheaper ones are executed first so that we can avoid expensive ones in case of `NO` decisions.